app/utils/answersHelpers/HolderOfTheTransitProcedureHelper.scala

```diff
@@ -0,0 +1,88 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils.answersHelpers
+
+import generated.AddressType10
+import models.reference.Country
+import models.{DynamicAddress, UserAnswers}
+import pages.holderOfTheTransitProcedure.CountryPage
+import play.api.i18n.Messages
+import uk.gov.hmrc.govukfrontend.views.viewmodels.summarylist.SummaryListRow
+import viewModels.sections.Section
+import viewModels.sections.Section.AccordionSection
+
+class HolderOfTheTransitProcedureHelper(
+  userAnswers: UserAnswers
+)(implicit messages: Messages)
+    extends AnswersHelper(userAnswers) {
+
+  def identificationNumber: Option[SummaryListRow] = buildRowWithNoChangeLink[String](
+    data = userAnswers.ie043Data.HolderOfTheTransitProcedure.flatMap(_.identificationNumber),
+    formatAnswer = formatAsText,
+    prefix = "unloadingFindings.rowHeadings.holderOfTheTransitProcedure.identificationNumber"
+  )
+
+  def tirHolderIdentificationNumber: Option[SummaryListRow] = buildRowWithNoChangeLink[String](
+    data = userAnswers.ie043Data.HolderOfTheTransitProcedure.flatMap(_.TIRHolderIdentificationNumber),
+    formatAnswer = formatAsText,
+    prefix = "unloadingFindings.rowHeadings.holderOfTheTransitProcedure.tirHolderIdentificationNumber"
+  )
+
+  def name: Option[SummaryListRow] = buildRowWithNoChangeLink[String](
+    data = userAnswers.ie043Data.HolderOfTheTransitProcedure.map(_.name),
+    formatAnswer = formatAsText,
+    prefix = "unloadingFindings.rowHeadings.holderOfTheTransitProcedure.name"
+  )
+
+  def address: Option[SummaryListRow] = buildRowWithNoChangeLink[DynamicAddress](
+    data = userAnswers.ie043Data.HolderOfTheTransitProcedure.map(_.Address.toDynamicAddress),
+    formatAnswer = formatAsDynamicAddress,
+    prefix = "unloadingFindings.rowHeadings.holderOfTheTransitProcedure.address"
+  )
+
+  def country: Option[SummaryListRow] = buildRowWithNoChangeLink[Country](
+    data = userAnswers.get(CountryPage),
+    formatAnswer = formatAsCountry,
+    prefix = "unloadingFindings.rowHeadings.holderOfTheTransitProcedure.country"
+  )
+
+  def holderOfTheTransitProcedureSection: Seq[Section] = userAnswers.ie043Data.HolderOfTheTransitProcedure
+    .map(
+      _ =>
+        AccordionSection(
+          sectionTitle = Some(messages("unloadingFindings.rowHeadings.holderOfTheTransitProcedure.heading")),
+          rows = Seq(
+            identificationNumber,
+            name,
+            country,
+            address,
+            tirHolderIdentificationNumber
+          ).flatten
+        )
+    )
+    .toList
+
+  implicit class RichAddress(address: AddressType10) {
+
+    def toDynamicAddress: DynamicAddress = DynamicAddress(
+      numberAndStreet = address.streetAndNumber,
+      city = address.city,
+      postalCode = address.postcode
+    )
+  }
```

Comment:
We have a few implicit classes for scalaxb classes in the `models` package object - maybe worth moving this there?